src/Integration/State/StateManager.cs

```diff
@@ -39,6 +40,7 @@ public StateManager(IHost host, TransferableVisualState state)
 
             this.Host = host;
             this.ManagedState = state;
+            this.isSolutionBound = this.ManagedState.HasBoundProject;
```

Comment:
FYI I added a method in ISolutionBindingInformationProvider - IsSolutionBound which should be used instead of reading the binding information directly. IsSolutionBound == true just means that the solution is bound, it could still be the case that GetBoundProjects will return an empty list (i.e. if the user deleted all the rulesets).